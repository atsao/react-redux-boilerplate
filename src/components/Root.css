@import '_sample';

body {
  font: 14px/1 Helvetica Neue, Arial, sans-serif;
  margin: 0;
  padding: 0;
}

a {
  color: inherit;
  text-decoration: none;
}

a:hover {
  text-decoration: underline;
}

.nav {
  margin-top: 8px;
}

.nav a {
  margin-right: 8px;
}

.header {
  background: #444;
  color: #fff;
  padding: 40px 24px;
}

.main-header {
  color: #fff;
  margin: 0;
}
