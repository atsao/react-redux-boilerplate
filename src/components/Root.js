import React, { Component, PropTypes } from 'react';

import '../assets/css/index.css';

class Root extends Component {
  render () {
    return (
      <div className="wrapper">
        <h1 className="example">Welcome to App</h1>
        {this.props.children}
      </div>
    );
  }
}

Root.propTypes = {
  children: PropTypes.node,
}

export default Root;
