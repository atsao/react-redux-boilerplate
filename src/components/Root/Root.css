.header {
  background: #444;
  color: #fff;
  padding: 40px 24px;
}

.main-header {
  color: #fff;
  margin: 0;
}
