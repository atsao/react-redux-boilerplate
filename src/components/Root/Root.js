import React, { Component } from 'react';
import PropTypes from 'prop-types';

import Nav from '../Nav';

import styles from './Root.css';
import '../../assets/css/index.css';

export class Root extends Component {
  render() {
    return (
      <div>
        <div className={styles.header}>
          <h1 className={styles['main-header']}>react-redux-boilerplate</h1>
          <Nav />
        </div>
        {this.props.children}
      </div>
    );
  }
}

Root.propTypes = {
  children: PropTypes.node
};

Root.defaultProps = {
  children: null
};

export default Root;
