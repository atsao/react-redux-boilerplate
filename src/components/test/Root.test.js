import React from 'react';
import { shallow } from 'enzyme';

import Root from '../Root';

describe('Root', () => {
  it('renders without crashing', () => {
    const tree = shallow(<Root />);
    expect(tree).toMatchSnapshot();
  });
});
