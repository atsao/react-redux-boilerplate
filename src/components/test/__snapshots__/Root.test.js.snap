// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Root renders without crashing 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <div
        className="header"
    >
        <h1
            className="main-header"
        >
            react-redux-boilerplate
        </h1>
        <Nav />
    </div>
</div>,
  "nodes": Array [
    <div>
      <div
            className="header"
      >
            <h1
                  className="main-header"
            >
                  react-redux-boilerplate
            </h1>
            <Nav />
      </div>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Root />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Root {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "children": null,
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div>
          <div
                    className="header"
          >
                    <h1
                              className="main-header"
                    >
                              react-redux-boilerplate
                    </h1>
                    <Nav />
          </div>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div>
          <div
                    className="header"
          >
                    <h1
                              className="main-header"
                    >
                              react-redux-boilerplate
                    </h1>
                    <Nav />
          </div>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Root />,
}
`;
